.github/workflows/main.yml

```
@@ -10,20 +10,20 @@ jobs:
+      - name: Log in to Docker Hub
+        uses: docker/login-action@0d4c9c5f114e0051d914bca15554477dd762a938  # v1.14.1  # v1.14.1
+        with:
         username: ${{ secrets.DOCKER_USERNAME }}
         password: ${{ secrets.DOCKER_PASSWORD }}
 
-    - name: Build and push
-      uses: docker/build-push-action@v2
-      with:
+      - name: Build and push
+        uses: docker/build-push-action@ad82d024503b15000a683bdffec2bb5c0ccca10c  # v2.10.0  # v2.10.0
+        with:
```

actionlint (1.7.12)

[error] "with" section should not be empty. please remove this section if it's unnecessary

(syntax-check)